src/debug/debugSession.ts

```diff
@@ -145,39 +146,53 @@ export class DebugSession implements IDebugSession {
 
         const { targetPod, targetPodNS, containers } = targetPodInfo;
 
-        // Select the image type to attach.
-        this.debugProvider = await this.pickupAndInstallDebugProvider();
-        if (!this.debugProvider) {
-            return;
-        }
-
         // Select the target container to attach.
         // TODO: consolidate with container selection in extension.ts.
         const targetContainer = await this.pickContainer(containers);
         if (!targetContainer) {
             return;
         }
 
-        // Find the debug port to attach.
-        const portInfo = await this.debugProvider.resolvePortsFromContainer(this.kubectl, targetPod, targetPodNS, targetContainer);
-        if (!portInfo || !portInfo.debugPort) {
-            await this.openInBrowser("Cannot resolve the debug port to attach. See the documentation for how to use this command.", debugCommandDocumentationUrl);
+        const processes = await getProcesses(this.kubectl, targetPod, targetPodNS, targetContainer);
+
+        // Select the image type to attach.
+        this.debugProvider = await this.pickupAndInstallDebugProvider(undefined, processes);
+        if (!this.debugProvider) {
             return;
         }
 
+        const supportedProcesses = processes ? this.debugProvider.filterSupportedProcesses(processes) : undefined;
+        const pidToDebug = supportedProcesses && supportedProcesses.length === 1 ? +supportedProcesses[0].pid : undefined;

```

Comment:
Suggest pulling out a function here with an intention-revealing name e.g. `extractTargetPID` or `findDebuggeePID` (of type `(ProcessInfo[] | undefined) => (number | undefined)`)

Reply:
fixed